### PR TITLE
Restream buffered frames with minimum publish rate

### PR DESCRIPTION
**Pull request by viktorku**
Image topics that are occasionally publishing end up buffered internally if not re-streamed at some rate. This PR adds an optional `publish_rate` parameter which if provided will make the server continuously restream messages older than the publish rate.

PS. We have been using this for many years at Pickit, albeit on an [old diverged fork](https://github.com/Intermodalics/web_video_server/tree/develop-minimum-rate-stream). This adapts some work @psoetens did four years ago, on top of the latest mainline version.

083332e444b495b7a5cae2079e26819046dc270a
2fb83e0d096f768e9d978c80d59ded62bcf45e37
71802876bf7a8170e92737eacf33a9b25724c91c
6ce0eed837b302efbf1d35ce41e8a6fa025cd0a4

### CODE REVIEW

File: src/ros_compressed_streamer.cpp
Comment (viktorku) [inline src/ros_compressed_streamer.cpp]: I'll follow up by implementing this overload as well

**Review by jihoonl — APPROVED** [review]

**Comment by jihoonl** [comment]
This cannot be merged until travis get fixed. This will be approved after #89 get merged.

**Comment by viktorku** [comment]
Merging this..

**Comment by msadowski** [comment]
@viktorku thanks a lot for this PR! It's very useful! I just tested it and it works perfectly! Are there any plans to release this to the debs any time soon?

**Comment by jihoonl** [comment]
@msadowski I just rolled the new version. It should be available in the next sync

**Comment by msadowski** [comment]
> @msadowski I just rolled the new version. It should be available in the next sync

Great, thanks!

**Pull request merged**
